ui/app/common/directives/pnc-select-items.js

```diff
@@ -0,0 +1,174 @@
+
+  var module = angular.module('pnc.common.directives');
+
+  /**
+   * @ngdoc directive
+   * @name pnc.common.directives:pncInfiniteSelectSingle
+   * @restrict E
+   * @param {string@} single-item
+   * Indicates if the list of selected items is single (selecting a new item will replace the existing one, if any). Defaults to 'false'
+   * @param {array=} selected-items
+   * An array on the in scope controller that will hold the items selected by
+   * the user. The array can be pre-populated to show items that are already
+   * selected.
+   * @param {string@} select-required
+   * Indicates if the select value is required (for UI validation). Defaults to 'false'
+   * @param {string@} select-id
+   * The ui id of the select directive (for UI validation)
+   * @param {string@} select-name
+   * The ui name of the select directive (for UI validation)
+   * @param {string@} placeholder
+   * The placeholder to show in the search text. Defaults to 'Scroll & Filter'
+   * @param {array=} items
+   * The items to display
+   * @param {string=} item-id
+   * The ui id of the items (for UI validation)
+   * @description
+   * A directive that allows users to select multiple options from a list of
+   * possible types. The user finds possible items by typing into a input box. 
+   * The selected items are presented to the user in a list (one-item list if 'single-item' is true)
+   * The user is given the option to remove items from the list by pressing a
+   * cross button next to the selected item.
+   * @example
+   * <pnc-select-items
+        single-item="true"
+        selected-items="ctrl.productSelection.selected"
+        select-required="true"
+        select-id="productId"
+        select-name="productId"
+        placeholder="Scroll & Filter Products..."
+        items="ctrl.products"
+        item-id="ctrl.selectedProductId">
+      </pnc-select-items>
+   * @author Andrea Vibelli
+   */
+  module.directive('pncSelectItems', function() {
+    return {
+      restrict: 'E',
+      scope: {
+        singleItem: '@',
+    	selectedItems: '=',
+        items: '=',
+        itemId: '=',
+        placeholder: '@',
+        selectId: '@',
+        selectName: '@',
+        selectRequired: '@'
+      },
+      templateUrl: 'common/directives/views/pnc-select-items.html',
+      controller: [
+        '$log',
+        '$scope',
+        function($log, $scope) {
+
+          var findInArray = function(obj, array) {
+            for (var i = 0; i < array.length; i++) {
+              if (angular.equals(obj.id, array[i].id)) {
+                return i;
+              }
+            }
+            return -1;
+          };
+
+       	  var PLACEHOLDER = 'Scroll & Filter';
+          $scope.placeholder = _.isUndefined($scope.placeholder) ? PLACEHOLDER : $scope.placeholder;
+          $scope.selectRequired = _.isUndefined($scope.selectRequired) ? false : $scope.selectRequired;
+          $scope.singleItem = convertToBooleanString($scope.singleItem);
```

Comment:
@matedo1 bad naming for the method, was meant to handle proper inittialization of the value, removed to an inline initialization. Thanks!